rust/gitxetcore/src/command/s3config.rs

```diff
@@ -0,0 +1,169 @@
+use crate::config::{get_global_config, XetConfig};
+use crate::errors::{GitXetRepoError, Result};
+use anyhow::anyhow;
+use clap::Args;
+use std::path::{Path, PathBuf};
+use xet_config::{Cfg, DEFAULT_XET_HOME};
+
+#[cfg(unix)]
+const S3_CONFIG_FILE_NAME: &str = "s3env";
+#[cfg(windows)]
+const S3_CONFIG_FILE_NAME: &str = "s3env.bat";
+
+#[derive(Args, Debug)]
+pub struct S3configArgs {
+    /// The domain for Xet S3 service
+    #[clap(long, default_value = "xethub.com")]
+    pub host: String,
+}
+
+pub fn s3config_command(config: XetConfig, args: &S3configArgs) -> Result<()> {
+    // login always write auth info into the global config.
+    let global_config = get_global_config()?;
+    let cfg = Cfg::from_file(&global_config).unwrap_or_default();
+
+    let mut config_ret = Err(GitXetRepoError::AuthError(anyhow!("No match profile")));
+
+    if args.host.is_empty() || args.host == "xethub.com" {
+        // this goes into the root profile
+        config_ret = write_xs3_config(&config, &cfg);
+    } else {
+        // search in the sub-profiles
+        for v in cfg.profiles.values() {
+            if let Some(e) = &v.endpoint {
+                if e == &args.host {
+                    config_ret = write_xs3_config(&config, v);
+                    break;
+                }
+            }
+        }
+    }
+
+    match config_ret {
+        Ok(path) => {
+            print_help_message(&path);
+            Ok(())
+        }
+        Err(GitXetRepoError::AuthError(_)) => {
+            print_auth_error_message(&args.host);
+            Ok(())
+        }
+        Err(GitXetRepoError::InvalidOperation(_)) => {
+            print_service_unavailable_message(&args.host);
+            Ok(())
+        }
+        Err(e) => Err(e),
+    }
+}
+
+fn print_help_message(s3config_path: &Path) {
+    #[cfg(unix)]
+    let default_path_str = format!("$HOME/{DEFAULT_XET_HOME}/{S3_CONFIG_FILE_NAME}");
+    #[cfg(windows)]
+    let default_path_str =
+        format!("%HOMEDRIVE%%HOMEPATH%\\{DEFAULT_XET_HOME}\\{S3_CONFIG_FILE_NAME}");
+    let path_str = s3config_path.to_str().unwrap_or(&default_path_str);
+    #[cfg(unix)]
+    eprintln!(
+        r#"Profile for the XetHub S3 Service is configured now. Great!
+
+1. To get started source the corresponding env file. This is usually done by
+running one of the following (note the leading DOT):
+. "{}"                            # For sh/bash/zsh/ash/dash/pdksh
```

Comment:
Can do `{path_str:<32}` to put path_str left-aligned and padded out with spaces to 32 character width if less. (noting the alignment above.